Add common project Makefiles

lclausen-adi opened the pull request:
The project Makefiles for share most of their code. The only difference is the list of project dependencies.

Create a file that has the common parts and can be included by the project
Makefiles.

This drastically reduces the size of the project Makefiles and also allows to change the Makefile implementation without having to re-generate all Makefiles.

acostina commented:
Given that for Xilinx all projects have a system_project.tcl, system_constr.xdc, system_bd.tcl and system_top.v, shouldn't we add those to common too ?

lclausen-adi commented:
Makes sense. Building would fail if these files don't exist, right?

Maybe even add scripts/adi_project.tcl, scripts/adi_env.tcl and scripts/adi_board.tcl as well.

lclausen-adi commented:
Not all have system_constr.xdc. E.g. the adv7511 projects.

Csomi commented:
I think just the adv7511/zc706 and adv7511/zed does not have one. We can fix it, and we should fix it.  All project should have an XDC file, even if it is empty.

acostina commented:
If we go with a common Make infrastructure, it should be ok to add 1-2 empty constraint files, if needed.  If the file don't exist, make will fail, yes.

lclausen-adi commented:
I can add a check to the Makefile and only add the file to the dependency list if it exists. (I think that is possible).

ronagyl commented:
Looks good.  Are we planning to do the same changes to the Makefiles from the library folder?

lclausen-adi commented:
Doing the same for library makes sense. I'll add that.

lclausen-adi commented:
@Csomi Some projects don't have a system_top.v, like ad7616_sdz.

lclausen-adi commented:
I can match on anything system_top*.v in the Makefile and add that as a dependency? Any potential problems with that?

lclausen-adi commented:
v2:
- Added dependencies on files that are used by (almost) all projects to the shared Makefiles
- Added shared Makefile for the library and re-generate all the IP core Makefiles

lclausen-adi commented:
v3:
- Add shared top-level project Makefile
- Small changes to the top-level Makefile
- Removed a few lines of just "##########..."
- Updated copyright year

lclausen-adi commented:
Can somebody give this a quick smoke test on Windows to see if there are any potential compatibility issues.

Csomi commented:
It seems to work in cygwin.

lclausen-adi commented:
Thanks.

lclausen-adi commented:
v4:
 - Small cleanups. Whitespace, variable naming, etc...
